src/extensions/renderer-api/catalog.ts

```diff
@@ -48,6 +48,33 @@ export class CatalogEntityRegistry {
   getItemsForCategory<T extends CatalogEntity>(category: CatalogCategory): T[] {
     return registry.getItemsForCategory(category);
   }
+
+  /**
+   * Add a onRun hook to a catalog entity.
+   * @param catalogEntityUid The uid of the catalog entity
+   * @param onRunHook The function that should return a boolean if the onRun of catalog entity should be triggered.
+   * @returns A function to remove that hook
+   */
+  addOnRunHook(catalogEntityUid: CatalogEntity["metadata"]["uid"], onRunHook: CatelogEntityOnRunHook) {
+    return registry.addOnRunHook(catalogEntityUid, onRunHook);
+  }
+
+  /**
+   * Returns one catalog entity onRun hook by catalog entity uid
+   */
+  getOnRunHook(catalogEntityUid: CatalogEntity["metadata"]["uid"]): CatelogEntityOnRunHook | undefined {
+    return registry.getOnRunHook(catalogEntityUid);
+  }
+
+  /**
+   * Returns all catalog entities' onRun hooks
+   */
+  getAllOnRunHooks(): Array<{
+    catalogEntityUid: CatalogEntity["metadata"]["uid"];
+    onRunHook: CatelogEntityOnRunHook;
+  }> {
+    return registry.getAllOnRunHooks();
+  }
```

Comment:
I don't think we need this API.

Reply:
Can you explain why? Intention: allow extension developer to check if hooks are already added.

Reply:
Why would it be the extension developer care? We should just use a `Set` then.

Reply:
At least in space extension we need this to check when the clusters are synced from the cloud.

Reply:
They should just add the handler again (using the built-in set type logic) or you should add a method to do that check. I don't see the point of this method in particular.

Reply:
realised we had `getOnRunHook` already, we can remove this.